source/contributing/documentation.md

```diff
@@ -7,6 +7,61 @@ Feel free to in touch with the [Nix documentation team](https://nixos.org/commun
 
 If you cannot contribute time, consider [donating to the NixOS Foundation's documentation project on Open Collective](https://opencollective.com/nixos/projects/documentation-project) to fund ongoing maintenance and development of reference documentation and learning materials.
 
+## Documentation framework
+We aim to build our documentation according to the [Diataxis][diataxis] framework, which divides documentation into four categories:
+
+[diataxis]: https://diataxis.fr
+
+- Reference
+- How-to
+- Explanation
+- Tutorial
+
+We've found that contributors struggle to understand the differences between these categories, and while we _strongly_ recommend reading up on the Diataxis framework, we can summarize them as follows:
+
+- Reference
+  - Reference material includes technical documentation, API documentation, etc.
+  - Reference material should be relatively terse, simply listing which functions/classes/etc exist, but should still clearly define how to use the code and how it relates to the rest of the ecosystem.
+  - Reference material should focus on "what's out there", with the text and layout optimized for scanning and random access.
+  - Reference material should be supplemented with examples.
+- How-to (we call these "Recipes")
+  - Recipes are a list of steps showing how to achieve a specific goal or solve a specific problem.
+  - The goal is to help the reader reach a specific end, not understand the theory or greater context.
+  - A recipe assumes that the reader already has the background to understand the topic at hand and therefore doesn't need to explain the introduction of each new concept.
+- Explanation (we call these "Concepts")
+  - Concepts describe the internals of a piece of code or how to think about a particular idea or entity in the ecosystem.
+  - A concept can also describe the historical context behind why something works the way that it does today.
+  - If you find yourself wanting to write about the nitty gritty details of how something works, you most likely want to write an explanation.
+- Tutorial
+  - Tutorials walk the user through a particular activity to help give them muscle memory with accepted tools and patterns in the ecosystem.
```

Comment:
```suggestion
  - Tutorials walk the user through a particular activity to help give them a first experience with accepted tools and patterns in the ecosystem.
```

Muscle memory would need repetition, which is not what we're going for necessarily.

Reply:
```suggestion
  - Tutorials walk the user through a particular activity to help give them a start with accepted tools and patterns in the ecosystem.
```